py/dml/compat.py

```diff
@@ -105,6 +105,19 @@ class port_obj_param(CompatFeature):
     last_api_version = api_5
 
 
+@feature
+class shared_logs_on_device(CompatFeature):
+    '''This compatibility feature changes the semantics of log statements
+    inside shared methods so that they always log on the device object, instead
+    of the nearest enclosing Simics object like with non-shared methods.
+    This behaviour was a bug present since the very introduction of shared
+    methods, which has lead to plenty of script code having become reliant
+    on it, especially in regards to how banks log. This feature preserves the
+    bugged behaviour.'''
+    short = "logs inside shared methods always log on the device object"
```

Comment:
Other `short` strings use imperative mood. OK to break the pattern if it reads better. (try --help-no-compat)